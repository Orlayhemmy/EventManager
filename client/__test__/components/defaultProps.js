export const props = {
  userState: {
    user: {
      isAdmin: true
    },
    isAuth: true
  },
  location: {
    pathname: '/'
  },
  center: {
    status: '',
    message: ''
  }
};

export const center = {
  status: 201,
  message: ''
};

export const bookedEvent = [
  { id: 1, eventTitle: 'Pizza Party', bookedDate: [], isApproved: true },
  { id: 2, eventTitle: 'Wedding Ceremony', bookedDate: [], isApproved: true  },
  { id: 3, eventTitle: 'Seminar', bookedDate: [], isApproved: false  }
];

export const logout = jest.fn();
export const onChange = jest.fn();
