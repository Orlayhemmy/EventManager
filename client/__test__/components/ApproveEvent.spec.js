import React from 'react';
import ApproveEvent from '../../components/CenterDetails/Template/Content/ApproveEvent';
import { onChange } from './defaultProps';

describe('Approve Event Component', () => {
  const onApprove = jest.fn();
  const showHiddenDiv = jest.fn();
  const state = {
    comment: ''
  }
  it('should render component without error', () => {
  const wrapper = renderer.create(<ApproveEvent onChange={onChange} onApprove={onApprove} showHiddenDiv={showHiddenDiv} state={state}/>).toJSON();
    expect(wrapper).toMatchSnapshot();
  })
})