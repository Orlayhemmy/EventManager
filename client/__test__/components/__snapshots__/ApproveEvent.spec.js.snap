// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Approve Event Component should render component without error 1`] = `
<div>
  <span>
    Are you sure you want to approve this event?
  </span>
  <div>
    <i
      className="fa fa-thumbs-up green"
      id="approve"
      onClick={[MockFunction]}
    />
    <i
      className="fa fa-thumbs-down trash"
      data-toggle-id="disapprove"
      onClick={[MockFunction]}
    />
    <br />
    <span>
      <br />
      Approve
    </span>
    <span>
      <br />
      Disapprove
    </span>
  </div>
  <div
    hidden={true}
    id="disapprove"
  >
    <p>
       
      Disapproved event will be deleted. Are you sure you want to disapprove event?
    </p>
    <div
      className="form-group"
    >
      <textarea
        className="form-control"
        id="comment"
        onChange={[MockFunction]}
        placeholder="Give reasons for disapproving this event"
        value=""
      />
    </div>
    <i
      className="fa fa-trash trash"
      id="disapprove"
      onClick={[MockFunction]}
    />
    <i
      className="fa fa-close"
      data-toggle-id="disapprove"
      onClick={[MockFunction]}
    />
    <br />
    <span>
      <br />
      delete
    </span>
    <span>
      <br />
      cancel
    </span>
  </div>
</div>
`;
